### PR TITLE
add Symfony binary

### PR DESCRIPTION
**Pull request by vasilvestre**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/winget-pkgs/pull/455)

### CODE REVIEW

File: manifests/Symfony/Binary/4.15.0.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Id: Symfony.Binary
```
Comment (fabpot) [inline manifests/Symfony/Binary/4.15.0.yaml:1]: Should be Symfony CLI instead

Comment (vasilvestre) [reply manifests/Symfony/Binary/4.15.0.yaml:1]: Fine thank you !

File: manifests/Symfony/CLI/4.15.0.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Id: Symfony.CLI
+Version: 4.15.0
+Name: CLI
+Publisher: Symfony
+License: © 2020 Symfony, Inc. All rights reserved.
+LicenseUrl: https://symfony.com/legal/cli-eula
+Tags: symfony, binary, cli
+Description: The symfony binary created by the installer provider the best way to create new Symfony applications, a powerful local web server to develop your projects and a loot to check for security vulnerabilities.
```
Comment (noniagriconomie) [inline manifests/Symfony/CLI/4.15.0.yaml:8]:
```suggestion
Description: The Symfony binary created by the installer provide the best way to create new Symfony applications, a powerful local web server to develop your projects and a tool to check for security vulnerabilities.
```

**Comment by KevinLaMS** [comment]
Still has SmartScreen Failure

**Comment by vasilvestre** [comment]
> Still has SmartScreen Failure

I've submitted the ticket during the night, might take long ?

**Comment by vasilvestre** [comment]
https://github.com/microsoft/winget-pkgs/pull/455#issuecomment-632402316
I got notification, latest définition now works !

**Comment by KevinLaMS** [comment]
Thanks.  I just tried it, but it did not work.  I will give it a few hours!!!!

**Comment by vasilvestre** [comment]
> Thanks. I just tried it, but it did not work. I will give it a few hours!!!!

>`Analyst comments:

We have removed the detection. Please follow the steps below to clear cached detection and obtain the latest malware definitions.

1. Open command prompt as administrator and change directory to c:\Program Files\Windows Defender
2. Run “MpCmdRun.exe -removedefinitions -dynamicsignatures”
3. Run "MpCmdRun.exe -SignatureUpdate"

Alternatively, the latest definition is available for download here: https://www.microsoft.com/en-us/wdsi/definitions`

**Comment by KevinLaMS** [comment]
still failing smart screen for me.

**Comment by vasilvestre** [comment]
I submit it in Smartscreen, seem strange

**Comment by vasilvestre** [comment]
> still failing smart screen for me.

And it's now done !

**Comment by denelon** [comment]
@vasilvestre I tried to install this in my test VM, and it failed with an exit code 3.  Are there any language dependencies or other application dependencies? I'm not familiar with symphony. 

I was also looking at the name and how the application would show up in search metadata. Would it be more clear to name the package `SymfonyCLI`, and the ID `Symfony.SymfonyCLI`? 

The good news is that I did not get the SmartScreen validation trigger.

**Comment by vasilvestre** [comment]
> @vasilvestre I tried to install this in my test VM, and it failed with an exit code 3. Are there any language dependencies or other application dependencies? I'm not familiar with symphony.
I think php may be.. 

> I was also looking at the name and how the application would show up in search metadata. Would it be more clear to name the package `SymfonyCLI`, and the ID `Symfony.SymfonyCLI`?
The foundator of the fw answered that CLI was nice to use. but I think SymfonyCLI will be cleaner. In winstall you see that name are really important for end-user.
 
> The good news is that I did not get the SmartScreen validation trigger.
Fine !

**Comment by KevinLaMS** [comment]
This app fails installation.  Because it appears to be just running, not actually installing.  And as a CLI it is unclear to me this will work for the user.

**Comment by vasilvestre** [comment]
> This app fails installation. Because it appears to be just running, not actually installing. And as a CLI it is unclear to me this will work for the user.

hmmm what do you recommand?

**Comment by vasilvestre** [comment]
> This app fails installation. Because it appears to be just running, not actually installing. And as a CLI it is unclear to me this will work for the user.

The tool is a CLI tool but that's actually a system-wide program to manage your app and even deploy it in the web. The install should make it available to the system.

**Comment by denelon** [comment]
@vasilvestre this PR is quite old. We've implemented a new schema for manifests. Would you like to update the PR, or close this request?

**Comment by vasilvestre** [comment]
> @vasilvestre this PR is quite old. We've implemented a new schema for manifests. Would you like to update the PR, or close this request?

Can we install binary now? :)

**Comment by denelon** [comment]
@vasilvestre does Symfony support silent installation, or does it display UI to the user before completing installation?

**Comment by vasilvestre** [comment]
> @vasilvestre does Symfony support silent installation, or does it display UI to the user before completing installation?

May be wrong but actual test on my local fails because the symfony binary ask for an update after the install

**Comment by arjun-msft** [comment]
The binary appears to be a standalone/portable exe and not an installer, support is being tracked via https://github.com/microsoft/winget-cli/issues/182

**Comment by KevinLaMS** [comment]
Clearing assignees until unblocked.

**Review by ItzLevvie — DISMISSED** [review]

**Comment by vasilvestre** [comment]
Validation now fails, is it due to new schema ?

**Comment by ItzLevvie** [comment]
> is it due to new schema ?

No it's not.

It's just that portable apps are not currently supported by WinGet yet but it'll be coming soon so that this PR can be unblocked:
image

**Comment by vasilvestre** [comment]
> > is it due to new schema ?
> 
> No it's not.
> 
> It's just that portable apps are not currently supported by WinGet yet but it'll be coming soon so that this PR can be unblocked: image image

Oh ok, as someone approved the PR I through it was ok :)

**Comment by Seb33300** [comment]
@vasilvestre Your PR is for Symfony cli 4 which is deprecated now.

Since version 5 Symfony cli is now open source and has a new repository (download url is different than version 4)
https://github.com/symfony-cli/symfony-cli

Can you please update your PR to install version 5

**Comment by vasilvestre** [comment]
> @vasilvestre Your PR is for Symfony cli 4 which is deprecated now.
> 
> Since version 5 Symfony cli is now open source and has a new GitHub repository (download url is different from version 4) https://github.com/symfony-cli/symfony-cli
> 
> Can you please update your PR to install version 5?

Sadly as long as Symfony use zip for the installer it's not possible to use winget. Scoop is an alternative tho

**Review by ItzLevvie — APPROVED** [review]

**Pull request closed without merging**